test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,178 @@
+#include <test/tools/ossfuzz/protomutators/YulProtoMutator.h>
+
+#include <libyul/Exceptions.h>
+
+#include <src/text_format.h>
+
+using namespace solidity::yul::test::yul_fuzzer;
+
+using namespace protobuf_mutator;
+
+using namespace std;
+
+template <typename Proto>
+using YPR = YulProtoCBRegistration<Proto>;
+using YPM = YulProtoMutator;
+using PrintChanges = YPM::PrintChanges;
+
+MutationInfo::MutationInfo(ProtobufMessage const* _message, string const& _info):
+	ScopeGuard([&]{ exitInfo(); }),
+	m_protobufMsg(_message)
+{
+	writeLine("----------------------------------");
+	writeLine("YULMUTATOR: " + _info);
+	writeLine("Before");
+	writeLine(SaveMessageAsText(*m_protobufMsg));
+
+}
+
+void MutationInfo::exitInfo()
+{
+	writeLine("After");
+	writeLine(SaveMessageAsText(*m_protobufMsg));
+}
+
+template <typename T>
+void YulProtoMutator::functionWrapper(
+	CustomFuzzMutator<T> const& _callback,
+	T* _message,
+	unsigned _seed,
+	unsigned _period,
+	string const& _info,
+	PrintChanges _printChanges)
+{
+	YulRandomNumGenerator random(_seed);
+
+	if (_seed % _period == 0)
+	{
+		if (_printChanges == PrintChanges::Yes)
+		{
+			MutationInfo m{_message, _info};
+			_callback(_message, random);
+		}
+		else
+			_callback(_message, random);
+	}
+}
+
+/// Add m/sstore(0, variable)
```

Comment:
Is this still true? The body seems to deal only with `sstore`

Reply:
> The body seems to deal only with sstore

I think you are referring to 

https://github.com/ethereum/solidity/pull/8305/files#diff-0a5c977e43dccf48918662e04ac26b74R60

`StoreFunc_Storage` is actually an enum that includes storage and memory ops
  - sstore
  - mstore
  - mstore8

The enum is defined here:

https://github.com/ethereum/solidity/blob/8fde9fd1c3723d230df9107d22cbfc16c15162d4/test/tools/ossfuzz/yulProto.proto#L240-L244

Reply:
ah ok!